staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -291,31 +296,29 @@ func (r *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{})
 	r.customStorageLock.Lock()
 	defer r.customStorageLock.Unlock()
 
-	// This can't work like this, but I just want to try something. Before there was
-	// some additional logic, but I got rid of it before pushing the commit.
-	// So, this is called when CRD is updated. That includes creation, but termination as well.
-	// Probably some logic like in the naming controller, but only to fire state change
-	// if CRD is creating for the first time.
-	if !apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Established) {
+	if !apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Established) &&
+		apiextensions.IsCRDConditionTrue(newCRD, apiextensions.Installing) {
+
 		establishedCondition := apiextensions.CustomResourceDefinitionCondition{
 			Type:    apiextensions.Established,
 			Status:  apiextensions.ConditionTrue,
 			Reason:  "InitialNamesAccepted",
 			Message: "the initial names have been accepted",
 		}
-		// This could miss some logic, like applying changes (present in the naming controller)
-		// Also, I'm supposed to update the newCRD, right?
+
+		// Update storage with new CRD condition.
 		apiextensions.SetCRDCondition(newCRD, establishedCondition)
 
-		// This is used to updated the server. The above `SetCRDCondition` only sets in storage.
-		//crdClient, err := internalclientset.NewForConfig(s.GenericAPIServer.LoopbackClientConfig)
-		//_, err = crdClient.CustomResourceDefinitions().UpdateStatus(newCRD)
-		//if err != nil {
-		//	glog.V(5).Info(err)
-		//	return
-		//}
-	}
+		// Update server with new CRD condition.
+		_, err := r.client.Apiextensions().CustomResourceDefinitions().UpdateStatus(newCRD)
+		if err != nil {
+			return
+		}
 
+		// Is something like this required? As we have already updated the local copy
+		// I think nothing is needed at this point.
+		//newObj = crd
```

Comment:
I don't think it matters much. If there is a change actually done, we will get another update (i.e. call here).